network-api/networkapi/wagtailpages/models.py

```diff
@@ -560,29 +574,63 @@ def filter_entries_for_tag(self, entries, context):
         specific model before we can test for whether i) there are tags
         to work with and then ii) those tags match the specified ones.
         """
-        type = self.filtered.get('type')
-        context['filtered'] = type
-
-        if type == 'tags':
-            terms = self.filtered.get('terms')
+        terms = self.filtered.get('terms')
+
+        # "unsluggify" all terms:
+        context['terms'] = [str(Tag.objects.get(slug=term)) for term in terms]
+
+        entries = [
+            entry
+            for
+            entry in entries.specific()
+            if
+            hasattr(entry, 'tags')
+            and not
+            # Determine whether there is any overlap between 'all tags' and
+            # the tags specified. This effects ANY matching (rather than ALL).
+            set([tag.slug for tag in entry.tags.all()]).isdisjoint(terms)
+        ]
 
-            # "unsluggify" all terms:
-            context['terms'] = [str(Tag.objects.get(slug=term)) for term in terms]
+        return entries
 
+    def filter_entries_for_category(self, entries, context):
+        """
+        NOTE: we currently assume only blog pages can have
+        categories. If that ever changes, we will need to rename
+        the model from BlogPageCategory to PageCategory and make
+        the corresponding adjustments to this code
+        """
+        slug = self.filtered.get('category')
+        cat = None
+        for bpc in BlogPageCategory.objects.all():
+            # We can't use .filter for @property fields,
+            # so we have to run through all categories =(
+            if bpc.slug == slug:
+                cat = bpc
+        
+        if cat is not None:
+            # make sure we bypass "x results for Y"
+            context['no_filter_ui'] = True
```

Comment:
I can't remember specifically, but should we set defaults for this? Does it not crash if you to access `no_filter_ui` in the template and it doesn't actually exist in the context dict? If not, ignore.

Reply:
no: as per Django convention, non-existent variables are treated as false